TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java

```diff
@@ -364,6 +387,40 @@ public void onClick(View view) {
         }
     }
 
+    public void onClickOkLoadOccupancyGridDialog(final String occupancyGridName) {
+        if (occupancyGridName == null || occupancyGridName.isEmpty()) {
+            Log.e(TAG, "Name is null or empty, unable to load the occupancy grid");
+            displayToastMessage(R.string.name_error);
+            return;
+        }
+        new AsyncTask<Void, Void, Void>() {
+            @Override
+            protected Void doInBackground(Void... params) {
+                mTangoServiceClientNode.callLoadOccupancyGridService(occupancyGridName);
+                return null;
+            }
+        }.execute();
+        mLoadOccupancyGridButton.setEnabled(false);
```

Comment:
Nit: first disable the button, then call the service. If something goes wrong while calling the service, the button should already be disabled.